pkg/api/validation/validation.go

```diff
@@ -2151,7 +2234,7 @@ func validateInitContainers(containers, otherContainers []api.Container, volumes
 	return allErrs
 }
 
-func validateContainers(containers []api.Container, volumes sets.String, fldPath *field.Path) field.ErrorList {
+func validateContainers(containers []api.Container, volumes sets.String, deviceVolumes []api.Volume, fldPath *field.Path) field.ErrorList {
```

Comment:
Do we still need deviceVolumes as an argument?

Reply:
I think so, you asked that I don't use a similar volumes list like is already passed in (volumes list is used by ValidateVolumeMounts) and that I also do the analysis in VaidateVolumeDevices...so I think I need to pass in the actual volumes don't I or can I get those another way so I can see  each pods attached volumes and VolumeSources?

Reply:
Oh I see now.  You need the actual Volume object to be able to validate the source type.  Can get rid of the `volumes` argument then?

Reply:
yep, I think so, if I refactor ValidateVolumeMounts to use []api.Volume reather than volume sets